pkg/controllers/managementuser/rbac/podsecuritypolicy/version_test.go

```diff
@@ -0,0 +1,144 @@
+package podsecuritypolicy_test
+
+import (
+	"fmt"
+	"testing"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/controllers/managementuser/rbac/podsecuritypolicy"
+	"github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3/fakes"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/version"
+)
+
+func newClusterLister(kubernetesVersion string) *fakes.ClusterListerMock {
+	return &fakes.ClusterListerMock{
+		GetFunc: func(namespace, name string) (*v3.Cluster, error) {
+			if name == "invalid" {
+				return nil, fmt.Errorf("invalid cluster: %s", name)
+			} else if name == "not ready" {
+				return &v3.Cluster{Status: v3.ClusterStatus{}}, nil
+			} else {
+				return &v3.Cluster{Status: v3.ClusterStatus{Version: &version.Info{GitVersion: kubernetesVersion}}}, nil
+			}
+		},
+	}
+}
+
+func TestCheckClusterVersionFailsForVersionsThatCannotBeParsed(t *testing.T) {
+	t.Parallel()
+	tests := []string{"", "⌘⌘⌘", "v1.2", "v1.24", "v1.24.a", "1.24", "1.24.a"}
+	for _, v := range tests {
+		v := v
+		t.Run(v, func(t *testing.T) {
+			t.Parallel()
+			clusterLister := newClusterLister(v)
+			err := podsecuritypolicy.CheckClusterVersion("test", clusterLister)
+			require.Error(t, err)
+			require.NotErrorIs(t, err, podsecuritypolicy.ErrClusterVersionIncompatible)
+		})
+	}
+}
+
+func TestCheckClusterVersionInspectsValidVersionsForCompatibilityWithPSP(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		version string
+		wantErr bool
+	}{
+		// regular version strings
+		{
+			version: "1.24.9",
+			wantErr: false,
+		},
+		{
+			version: "v1.24.9",
+			wantErr: false,
+		},
+		{
+			version: "1.25.9",
+			wantErr: true,
+		},
+		{
+			version: "v1.25.9",
+			wantErr: true,
+		},
+		{
+			version: "v1.26.9",
+			wantErr: true,
+		},
+		// k3s version strings
+		{
+			version: "v1.24.9+k3s1",
+			wantErr: false,
+		},
+		{
+			version: "v1.25.9+k3s1",
+			wantErr: true,
+		},
+		{
+			version: "v1.26.9+k3s1",
+			wantErr: true,
+		},
+		// rke1 version strings
+		{
+			version: "v1.24.9-rancher1-1",
+			wantErr: false,
+		},
+		{
+			version: "v1.25.9-rancher1-1",
+			wantErr: true,
+		},
+		{
+			version: "v1.26.9-rancher1-1",
+			wantErr: true,
+		},
+		// rke2 version strings
+		{
+			version: "v1.24.9+rke2r1",
```

Comment:
I see we have examples for the rke2/k3s version strings - just to confirm, are the EKS/GKE/AKS version strings just the basic version or do they have a more complex syntax as well? What about generic imported clusters?

Reply:
I tried creating a managed DigitalOcean cluster and saw that its version on the DO portal appears to be `1.26.9-do.0`.
But when I import it into Rancher, I see it as `1.26.9` in Rancher. I will test other imported clusters.